workflows/differentialabundance.nf

```diff
@@ -157,8 +177,35 @@ workflow DIFFERENTIALABUNDANCE {
         ch_versions = ch_versions
             .mix(AFFY_JUSTRMA_RAW.out.versions)
 
-    }
-    else if(params.study_type == 'geo_soft_file'){
+    } else if (params.study_type == 'maxquant'){
+
+        // Split contrasts for proteus and for the later modules
+        ch_contrasts_split = ch_contrasts_file
+            .splitCsv ( header:true, sep:(params.contrasts.endsWith('tsv') ? '\t' : ','))
+            .map{ it.tail().first() }
+
+        // For proteus, extract only contrast variable as the module has to run once per contrast
+        ch_contrasts_proteus = ch_contrasts_split
+            .map{
+                tuple('id': it.variable)
+            }
+            .unique()
+
+        // Run proteus to import protein abundances
+        PROTEUS(
+            ch_contrasts_proteus.combine(proteus_in)
+        )
+
+        // Re-map the proteus output tables to the study ID as the tables are the same across contrasts
+        ch_in_raw = PROTEUS.out.raw_tab
+            .reduce{a, b -> a}
+            .map{tuple('id': exp_meta.id, it[1])}
+        ch_in_norm = PROTEUS.out.norm_tab
+            .reduce{a, b -> a}
+            .map{tuple('id': exp_meta.id, it[1])}
```

Comment:
```suggestion
        ch_in_raw = PROTEUS.out.raw_tab
            .first()
            .map{ meta, matrix -> tuple(exp_meta, matrix) }
            
        ch_in_norm = PROTEUS.out.norm_tab
            .first()
            .map{ meta, matrix -> tuple(exp_meta, matrix) }
```

I think this is much simpler to understand.

Reply:
Oop, I did not realize `first()` has the same effect as my `reduce()` snippet. Thanks!